src/routing/uploadGCPRouter.js

```diff
@@ -0,0 +1,52 @@
+import Router from 'koa-router';
+import uuidV4 from 'uuid/v4';
+import Storage from '@google-cloud/storage';
+
+export default class GoogleUploadRouter extends Router {
+  //pass in googleConfig, trace this back 
+  constructor(googleConfig) {
+    console.log(googleConfig)
+    super();
+    this.bucket = googleConfig.GCP_Bucket;
+    this.storage = new Storage({
+      credentials: {
+        "client_email": googleConfig.client_email,
+        "private_key": googleConfig.private_key,
+      }
+    });
+
+    this.prefix('/upload')
+      .post('/signRequest', this.getSignedRequest);
+  }
+
+  getSignedRequest = async ctx => {
+    const body = ctx.request.body;
+    const fileId = body.fileId || uuidV4();
+    let folder = body.folder && `${body.folder}/` || '';
+    try {
+      let url = await this.storage
+        .bucket(this.bucket)
+        .file(`${folder}${fileId}.jpg`)
+        .getSignedUrl({
+          action: 'write',
+          expires: '08/22/2019',
```

Comment:
Static date here is bad.  Make it like 1 hour from now, maybe 24 at most.  having an open write url is dangerous.  Also this will break in a year and people will be like WTF